pypeit/core/coadd.py

```diff
@@ -567,6 +564,31 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     flux_med, ivar_med = median_filt_spec(flux, ivar, mask, med_width)
     flux_ref_med, ivar_ref_med = median_filt_spec(flux_ref, ivar_ref, mask_ref, med_width)
 
+    if 'poly' in model:
+        #guess = np.append(ratio, np.zeros(norder))
+        yval = flux_ref_med
```

Comment:
Did some more digging around here with coadding, and I think that `yval` here needs to be something like `flux_ref_med/flux_med` since in `poly_ratio_fitfunc` we are multiplying the results of the model fit by `flux`.

Reply:
Can you confirm that this works well now?